crates/relayer-cli/src/commands/evidence.rs

```diff
@@ -0,0 +1,651 @@
+                &config,
+                &counterparty_chain_id,
+                rt.clone(),
+                |chain_config| {
+                    if let Some(key_name) = key_name {
+                        chain_config.key_name = key_name.to_string();
+                    }
+                },
+            )?;
+
+            chains.insert(counterparty_chain_id.clone(), chain_handle);
+        };
+
+        let counterparty_chain_handle = chains.get(&counterparty_chain_id).unwrap();
+
+        let signer = counterparty_chain_handle.get_signer()?;
+
+        // If the misbehaving chain is a CCV consumer chain,
+        // then try fetch the consumer chains of the counterparty chains.
+        // If that fails, then that chain is not a provider chain.
+        // Otherwise, check if the misbehaving chain is a consumer of the counterparty chain,
+        // which is definitely a provider.
+        let counterparty_is_provider = if chain.config().ccv_consumer_chain {
+            let consumer_chains = counterparty_chain_handle
+                .query_consumer_chains()
+                .unwrap_or_default(); // If the query fails, use an empty list of consumers
+
+            // FIXME: Do do we need to check if the client id also matches?
+            // ie. is it okay to submit a `MsgSubmitIcsConsumerDoubleVoting` to all clients
+            // of the provider chain, or should we only do this for the CCV client, and
+            // use the standard message for other clients?
+            consumer_chains
+                .iter()
+                .any(|(chain_id, _)| chain_id == chain.id())
+        } else {
+            false
+        };
+
+        if !counterparty_is_provider {
+            debug!("counterparty chain {counterparty_chain_id} is not a CCV provider chain, skipping...");
+            continue;
+        }
+
+        info!("submitting CCV misbehaviour to provider chain {counterparty_chain_id}");
+
+        // Construct the light client block header for the consumer chain at the infraction height
+        let infraction_block_header = {
+            let infraction_height = dv.vote_a.height;
+
+            let trusted_validator_set = rt
+                .block_on(chain.rpc_client.validators(infraction_height, Paging::All))?
+                .validators;
+
+            let signed_header = rt
+                .block_on(chain.rpc_client.commit(infraction_height))?
+                .signed_header;
+
+            let validators = rt
+                .block_on(chain.rpc_client.validators(infraction_height, Paging::All))?
+                .validators;
+
+            let validator_set =
+                validator::Set::with_proposer(validators, signed_header.header.proposer_address)?;
+
+            TendermintHeader {
+                signed_header,
+                validator_set,
+                trusted_height: Height::from_tm(infraction_height, chain.id()),
```

Comment:
FIXME: This seems wrong, as the light client will reject a header that has the same trusted height as its header, ie. if both are at `infraction_height` (see line 286 above). Though that hasn't happened in practice so far as far as I can tell, which is weird no?

cc @ancazamfir

Reply:
Maybe the on-chain misbehavior module does not verify that header with the light client when it gets it as evidence of a duplicate vote?

Reply:
yes, we need to get the trusted height in the same way we do for client updates. Retrieve the consensus state at the highest height smaller than the infraction height.

Reply:
Right, duh. Thank you, will get to that!

Reply:
Fixed in [5ff700e](https://github.com/informalsystems/hermes/pull/3456/commits/5ff700e3faaf6cbe42f53ddfc7896000d36de7f2) and cleaned up in [4762131](https://github.com/informalsystems/hermes/pull/3456/commits/47621312d2f075ac589d02eb115487a26cc19b47)